### ISSUE TITLE
cmd/gc: compiling error in 1.4 but not in 1.5

### ISSUE THREAD

**Issue by yangzhe1991** [golang/go#12927]
There is a strange compiling error on ugorji/go/codec's latest commit, can see https://github.com/ugorji/go/issues/108 for details. It only shows in 1.4, go 1.5 can compile without any error.

**Comment by hkjn** [comment]
To add some details directly in this report (thanks for filing it, @yangzhe1991), I sent the following in an email to golang-nuts (but I believe it might have gotten stuck in the moderation queue):

As a repro case, if you have Docker installed, this fails at time of writing:
```
docker run -it golang:1.4.2 go get github.com/ugorji/go/codec
# github.com/ugorji/go/codec
src/github.com/ugorji/go/codec/fast-path.generated.go:1824: v declared and not used
```

Our team (and others) noticed failing builds after commit [5d8e7](https://github.com/ugorji/go/commit/5d8e71f2040f1161ca038d1f91f9dcfd5e277294), but the line indicated seems to actually use "v" just after its declaration.

The issue is not present in Go 1.5.1:
```
docker run -it golang:1.5.1 go get github.com/ugorji/go/codec # no error
```

I've asked on the issue if the maintainer would be willing to rollback the commit since we and many others have broken builds at current time, but I wanted to send off this quick thread to see if others have thoughts on what's going on, or if there's a bug report filed yet for this. (Yes, vendoring would prevent this.. yes, we should and would like to be on Go 1.5.)

**Comment by ianlancetaylor** [comment]
It seems to have to do with the number of cases in the type switch.

Here is a standalone test case: http://play.golang.org/p/kUOKZGIK4o .

If I remove even one case, the file compiles.

I suppose we should consider fixing this if we ever make a Go 1.4.4 release.  Otherwise, since it works in Go 1.5 and tip, there is nothing to do.

**Comment by josharian** [comment]
A fair amount of time has passed, so I'm checking in--does anyone still care whether this gets fixed on the 1.4 branch?

**Comment by hkjn** [comment]
@josharian: Not me, at least, I've long since moved to 1.5+ for everything I care about.

**Comment by josharian** [comment]
Thanks for the speedy response. I'm going to tentatively move this to Unplanned.

**Comment by ianlancetaylor** [comment]
There is no point to leaving this as unplanned.  We should either fix it in a 1.4.4 release, or close it.  Since there doesn't seem to be a reason to fix in 1.4.4, closing.
